src/components/ParserComponent.tsx

```diff
@@ -70,15 +72,93 @@ export default class ParserComponent extends React.Component<ParserInterface,
     return true;
   }
 
+  /**
+   * Purpose: to instantiate an empty list of objects
+   * for tracking the kinds of data in a column
+   * @param: fieldLength the number of columns of data
+   * @return: a list of objects
+   */
+  // eslint-disable-next-line max-len,require-jsdoc
+  createTypeCountingObjects(fieldLength: number) : CountTypes[] {
+    const typesForEachCol = [];
+    // instantiate object for each column
+    for (let i = 0; i < fieldLength; i++) {
+      typesForEachCol.push(new CountTypes());
+    }
+    return typesForEachCol;
+  }
+
   /**
    * Purpose: attempts to infer the types of the data in each of the columns
    * of the csv data
-   * @param {Array} data: the array of data to infer the types for
-   * @return {Array}: a list of objects which define the methods available for
-   * the data
+   * @param {Array} data: the array of pre-sorted valid data
+   * @return {Array}: a list of objects of type Column
    */
-  inferTypes(data: Array<object>): Array<object> {
-    return [];
+  inferTypes(data: Array<object>): Array<Column> {
+    if (this.state.data.length > 0) {
+      const listFields = Object.keys(this.state.data[0]);
+      // instantiate objects to track the types of data
+      const typesForEachCol = this.createTypeCountingObjects(listFields.length);
+      // check half the values to find if the data is consistent
+      [0, Math.floor(this.state.data.length / 2)].forEach((element) => {
+        const row: object = this.state.data[element];
+        // look at each field and categorize
+        for (let i = 0; i < listFields.length; i++) {
+          const curColTypes = typesForEachCol[i];
+          try {
+            // @ts-ignore
+            const val = row[listfields[i]];
+            const date = moment(val, 'YYYY-MM-DD');
+            const isValid = date.isValid();
+            if (isValid) {
+              curColTypes['numDate'] += 1;
+            }
+          } catch {
+            // @ts-ignore
+            const type = typeof row[listFields[i]];
+            if (type !== 'string' && type !== 'number') {
+              curColTypes['numIncongruent'] += 1;
+            }
+            // logs all the types that are seen
+            if (type === 'string') {
+              curColTypes['numString'] += 1;
+            } else {
+              curColTypes['numNumber'] += 1;
+            }
+          }
+        }
+      });
+      let indx = 0;
+      const arrayOfColumns = new Array<Column>(listFields.length);
+      typesForEachCol.forEach((element) => {
+        // checks the most common type and uses that
+        const mostCommonType = element.largest();
+        let newCol: Column;
+        if (mostCommonType === 'string') {
+          // create a Column object with occurrence data
+          // eslint-disable-next-line max-len
```

Comment:
I see this used a lot, can't we just split the line?

```
newCol = new Column(mostCommonType, enumDrawType.occurence,
                                                                   listFields[indx]);
``

Reply:
Okay github formats it retardedly but you get the idea

Reply:
Yup, can do